ast/ast.go

```diff
@@ -36,6 +36,10 @@ type FieldExpr interface {
 	Copy() FieldExpr
 }
 
+type Expression interface {
+	exprNode()
```

Comment:
This is missing `Copy()` ... I think we should either remove it everywhere, or add it here. 

I think it's useful to have, but maybe we should switch the implementation to use some deep copy library (or roll our own with reflect).

Reply:
Since we're not actually using `Copy()` anywhere, I'm loathe to spend time on this (and associated tests).  Since we support json serialization and deserialization, how about if somebody suddenly has an urgent need to make a deep copy of an ast, they just make a round trip through json?

Reply:
That seems like a good approach too. The point remains I think that we should either have `Copy()` on all nodes, or on none...

Reply:
@nwt you were the other one who objected to removing `Copy()` are you okay with using json if/when we ever need to copy asts?
FWIW, the immediate problem with adding it here is that FieldRead wants to be both a FieldExpr and an Expression so those two interfaces can't both have `Copy()` methods with different signatures...

Reply:
> FWIW, the immediate problem with adding it here is that FieldRead wants to be both a FieldExpr and an Expression so those two interfaces can't both have Copy() methods with different signatures...

Ugh. `Copy()` should go! :)

Reply:
Don't care.